File: Tribonacci Sequence.js
function tribonacci(signature, n) {
  let add, inc;
  let seq = [];

  if (n < 1) {
    return [];
  }

  if (n < signature.length) {
    return signature.slice(0, n);
  }

  inc = 0;
  seq = signature.slice(0);

  while (inc <= n) {
    add = seq.slice(inc, inc + 3).reduce((a, b) => b + a, 0);
    seq.push(add);
    inc += 1;
  }

  return seq.slice(0, n);
}

console.log(tribonacci([1, 1, 1], 10));

File: Tribonacci Sequence.py
def tribonacci(signature, n):
    
    # if less than 1, return a blank list
    if n<1:
        return []
    # if `n` is less than the signature,
    # return a list at the item's place
    if n<len(signature):
        return signature[0:n]
    
    # counter
    inc = 0
    
    # copy the signature list as a starting point
    seq = signature[:]
    
    # increment
    while inc<=n:
        # add up the last 3 items
        add = sum(seq[inc:inc+3])
        # add to the new list
        seq.append(add)
        # next!
        inc += 1
        
    # return the new list,
    # forcing to only the max size wanted
    return seq[0:n]
  
print(tribonacci([1, 1, 1], 10))
